### DISCUSSION TITLE
How to clear storage

### DISCUSSION THREAD

**Discussion by occuros** [emilk/egui#1698] [Ideas]
The app stage can be conveniently stored through the persistence feature.

It would be great to give the option to clear the stored data (and relaunch the app) in case any issue is encountered due to inconsistent storage data (due to a crash or panic).

The stored data can be manually deleted while going to the file directory, but it would be ideal if that could also be exposed directly in the UI.

**Comment by emilk** [comment]
There are three things that are persisted:

* egui memory
* App state
* native window state

Clearing the egui memory is easy: `*egui_ctx.memory() = Default::default();`

Clearing the app state you should be able to do yourself (`*self = MyApp::default();`).

Saving the new (cleared) state will be done automatically on shutdown (and every 30 seconds during autosave).
